src/screens/HomeScreen.js

```diff
@@ -0,0 +1,242 @@
+import React from "react";
+import "../App.css";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import DUMMY_DATA from "../data/boards.json";
+import DUMMY_CARDS from "../data/card.json";
+import NewBoardForm from "../components/NewBoardForm";
+import BoardsList from "../components/BoardsList";
+import CardsForSelectedBoard from "../components/CardsForSelectedBoard";
+import CreateNewCard from "../components/CreateNewCard";
+import SortOption from "../components/SortOption";
+
+const kBaseUrl = process.env.REACT_APP_BACKEND_URL;
+
+const transformCardResponse = (card) => {
+  const {
+    id,
+    message,
+    likes_count: likesCount,
+    board_id: boardId,
+    status,
+  } = card;
+  return { id, message, likesCount, boardId, status };
+};
+
+const INIT_DATA = DUMMY_DATA.map((board) => {
+  const reformedCards = board.cards.map((card) => transformCardResponse(card));
+  return { ...board, cards: reformedCards };
+});
+const INIT_CARDS = DUMMY_CARDS.map((card) => transformCardResponse(card));
+
+const getAllBoards = () => {
+  return axios
+    .get(`${kBaseUrl}/boards`)
+    .then((response) => {
+      return response.data;
+    })
+    .catch((error) => {
+      console.log(error);
+      return INIT_DATA; //this will be deleted after connecting with Back-end
+    });
+};
+
+const getCardsForSelectedBoard = (id) => {
+  return axios
+    .get(`${kBaseUrl}/boards/${id}/cards`)
+    .then((response) => {
+      console.log(response.data.map(transformCardResponse));
+      return response.data.map(transformCardResponse);
+    })
+    .catch((error) => {
+      console.log(error);
+      return INIT_CARDS.filter((card) => card.boardId === id);
+    });
+};
+
+const HomeScreen = () => {
+  const [cardData, setCardData] = useState([]);
+  const [boardsList, setBoardList] = useState(INIT_DATA);
+
+  const fetchBoards = () => {
+    getAllBoards().then((boards) => {
+      setBoardList(boards);
+    });
+  };
+
+  useEffect(() => {
+    fetchBoards();
+  }, []);
+
+  const fetchCards = (id) => {
+    getCardsForSelectedBoard(id).then((cards) => {
+      setCardData(cards);
+    });
+  };
+
+  const handleUpdatedBoard = (newBoard) => {
+    if (!newBoard.title || !newBoard.owner) {
+      alert("Please enter a title and owner!");
+      return;
+    }
+
+    console.log(newBoard);
+    axios
+      .post(`${kBaseUrl}/boards`, {
+        title: newBoard.title,
+        owner: newBoard.owner,
+      })
+      .then((response) => {
+        return response.data;
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+
+    setBoardList((boardsList) => {
+      const newBoardsList = [...boardsList];
+      newBoardsList.push({ ...newBoard, id: boardsList.length + 1 });
+      return newBoardsList;
+    });
```

Comment:
Because this does not happen in a `.then` block, it will run while the axios call is still ongoing. If there is an error with saving the board to the database, the error will be printed in the console, but unless the user has the console open, they might never know their board isn't actually in the remote database and be confused if they come back later and don't see it.